### PR TITLE
fix: manual biome linter errors

### PR DESCRIPTION
**Pull request by afonsojramos**

### CODE REVIEW

**Review by setchy — CHANGES_REQUESTED** [review]
Keen to learn more about biome

File: src/components/fields/FieldInput.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { FC, ReactNode } from "react";
+import type { FC, ReactNode } from "react";
```
Comment (setchy) [inline src/components/fields/FieldInput.tsx:1]: Interesting, do you know why biome suggests this to be imported?

File: src/routes/Settings.tsx [hunk 2]
```diff
@@ -211,15 +212,17 @@ export const SettingsRoute: FC = () => {
       </div>
 
       <div className="flex justify-between items-center bg-gray-200 dark:bg-gray-darker py-1 px-8">
-        <small
```
Comment (setchy) [inline src/routes/Settings.tsx:214]: Is biome suggesting this should be removed?

File: src/utils/theme.ts [hunk 3]
```diff
@@ -25,10 +25,7 @@ export const setTheme = (mode?: Theme) => {
       break;
 
     default:
-      if (
-        window.matchMedia &&
-        window.matchMedia("(prefers-color-scheme: dark)").matches
-      ) {
+      if (window.matchMedia?.("(prefers-color-scheme: dark)").matches) {
```
Comment (setchy) [inline src/utils/theme.ts:28]: Love it!

File: src/app.tsx [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { useContext } from "react";
 import {
-  HashRouter as Router,
   Navigate,
   Route,
+  HashRouter as Router,
```
Comment (setchy) [inline src/app.tsx:5]: Been really wanting sorted imports, thank you!

File: src/components/NotificationRow.tsx [hunk 5]
```diff
@@ -101,13 +105,14 @@ export const NotificationRow: FC<IProps> = ({ notification, hostname }) => {
               <>
                 {" "}
                 by{" "}
-                <span
+                <button
+                  type="button"
                   className="cursor-pointer"
                   title="View User Profile"
                   onClick={openUserProfile}
```
Comment (setchy) [inline src/components/NotificationRow.tsx:112]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/NotificationRow.tsx [hunk 6]
```diff
@@ -116,6 +121,7 @@ export const NotificationRow: FC<IProps> = ({ notification, hostname }) => {
 
       <div className="flex justify-center items-center gap-2 opacity-0 group-hover:opacity-80 transition-opacity">
         <button
+          type="button"
           className="focus:outline-none h-full hover:text-green-500"
           title="Mark as Done"
           onClick={() => markNotificationDone(notification.id, hostname)}
```
Comment (setchy) [inline src/components/NotificationRow.tsx:127]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/NotificationRow.tsx [hunk 7]
```diff
@@ -124,6 +130,7 @@ export const NotificationRow: FC<IProps> = ({ notification, hostname }) => {
         </button>
 
         <button
+          type="button"
           className="focus:outline-none h-full hover:text-red-500"
           title="Unsubscribe"
           onClick={unsubscribe}
```
Comment (setchy) [inline src/components/NotificationRow.tsx:136]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/NotificationRow.tsx [hunk 8]
```diff
@@ -132,6 +139,7 @@ export const NotificationRow: FC<IProps> = ({ notification, hostname }) => {
         </button>
 
         <button
+          type="button"
           className="focus:outline-none h-full hover:text-green-500"
           title="Mark as Read"
           onClick={() => markNotificationRead(notification.id, hostname)}
```
Comment (setchy) [inline src/components/NotificationRow.tsx:145]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/Repository.tsx [hunk 9]
```diff
@@ -38,23 +38,33 @@ export const RepositoryNotifications: FC<IProps> = ({
...
             {repoName}
           </span>
         </div>
 
         <div className="flex justify-center items-center gap-2 opacity-0 group-hover:opacity-80 transition-opacity">
           <button
+            type="button"
             className="focus:outline-none h-full hover:text-green-500"
             title="Mark Repository as Done"
             onClick={markRepoAsDone}
```
Comment (setchy) [inline src/components/Repository.tsx:70]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/Repository.tsx [hunk 10]
```diff
@@ -65,6 +75,7 @@ export const RepositoryNotifications: FC<IProps> = ({
           <div className="w-[14px]" />
 
           <button
+            type="button"
             className="focus:outline-none h-full hover:text-green-500"
             title="Mark Repository as Read"
             onClick={markRepoAsRead}
```
Comment (setchy) [inline src/components/Repository.tsx:81]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/Sidebar.tsx [hunk 11]
```diff
@@ -46,6 +46,7 @@ export const Sidebar: FC = () => {
     <div className="flex flex-col fixed left-14 w-14 -ml-14 h-full bg-gray-sidebar overflow-y-auto">
       <div className="flex flex-col flex-1 items-center py-4">
         <button
+          type="button"
           className="w-5 my-3 mx-auto cursor-pointer outline-none"
           title="Open Gitify on GitHub"
           onClick={onOpenBrowser}
```
Comment (setchy) [inline src/components/Sidebar.tsx:52]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/Sidebar.tsx [hunk 12]
```diff
@@ -73,6 +75,7 @@ export const Sidebar: FC = () => {
         {isLoggedIn && (
           <>
             <button
+              type="button"
               className={sidebarButtonClasses}
               title="Refresh Notifications"
               onClick={() => {
```
Comment (setchy) [inline src/components/Sidebar.tsx:81]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/Sidebar.tsx [hunk 13]
```diff
@@ -89,6 +92,7 @@ export const Sidebar: FC = () => {
             </button>
 
             <button
+              type="button"
               className={sidebarButtonClasses}
               title="Settings"
               onClick={() => {
```
Comment (setchy) [inline src/components/Sidebar.tsx:98]: Do we need to add an `onKeyDown` like the other changes in this PR?

File: src/components/fields/FieldInput.tsx (hunk 1 above)
Comment (afonsojramos) [reply src/components/fields/FieldInput.tsx:1]: This is not an import, this is marking an import as a type import only, read more here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

**Review by setchy — APPROVED** [review]

**Comment by afonsojramos** [comment]
image
Stacked PRs cannot be squash merged.... Oops...

**Comment by afonsojramos** [comment]
Should be good now, just self-reviewed the PR

**Comment by bmulholland** [comment]
If you use Graphite.dev, it will squash merge PRs for you.

**Pull request merged**